core/src/context.rs

```diff
@@ -1,42 +1,50 @@
-use std::collections::HashMap;
-
-use parser::Element;
+use crate::Element;
+use parser::ModuleArguments;
+use serde::{Deserialize, Serialize};
+use std::{
+    collections::HashMap,
+    io::{Read, Write},
+};
 #[cfg(feature = "native")]
 use wasmer::Cranelift;
-use wasmer::Store;
+use wasmer::{Instance, Store};
+use wasmer_wasi::{Pipe, WasiState};
 
-use crate::{CoreError, NodeName, Package, PackageInfo, Transform};
+use crate::{ArgInfo, CoreError, OutputFormat, Package, PackageInfo, Transform};
 
 #[derive(Debug)]
 pub struct Context {
-    packages_by_transform: HashMap<Transform, Package>,
-    packages_by_name: HashMap<String, Package>,
-    all_transforms_for_node: HashMap<NodeName, Vec<(Transform, Package)>>,
+    packages: HashMap<String, Package>,
+    transforms: HashMap<(String, OutputFormat), (Transform, Package)>,
     store: Store,
 }
 
 impl Context {
     pub fn new() -> Self {
         Context {
-            packages_by_transform: HashMap::new(),
-            all_transforms_for_node: HashMap::new(),
-            packages_by_name: HashMap::new(),
+            packages: HashMap::new(),
+            transforms: HashMap::new(),
             store: get_new_store(),
         }
     }
 
     pub fn load_default_packages(&mut self) {
         self.load_package(include_bytes!(concat!(
             env!("OUT_DIR"),
-            "/test-module/wasm32-wasi/release/test-module.wasm"
+            "/table/wasm32-wasi/release/table.wasm"
```

Comment:
Seems like a fine way to do it for now, but it would maybe be nice to make a macro for loading to decrease repetition

Reply:
Agree! But I think we can add that later : )